tests/integration/adapter.test.ts

```diff
@@ -0,0 +1,37 @@
+import Elder, { PostgresAdapter, Model, type } from '../../src'
+
+const config = { database: 'ash' }
+class CatModel extends Model {
+  @type('string') name: string
+}
+
+describe('Adapter', () => {
+  test('check connection works', async () => {
+    const adapter = PostgresAdapter.create(config)
+    await adapter.checkConnection()
+    return adapter.destroy()
+  })
+
+  test('check connection throws', async () => {
+    const adapter = PostgresAdapter.create({ database: 'doesnotexist' })
+    try {
+      await adapter.checkConnection()
+    } catch (err) {
+      expect(err).toBeInstanceOf(Error)
+    }
+    return adapter.destroy()
+  })
+
+  test('connection teardown', async () => {
+    const adapter = PostgresAdapter.create(config)
+    return adapter.destroy()
+  })
+
+  test('check basic functionality', async () => {
+    const adapter = PostgresAdapter.create(config)
+    const cats = await adapter.all(CatModel)
+
+    expect(cats[0].name).toBe('fluffy')
+    return adapter.destroy()
+  })
+})
```

Comment:
The adapter is the workhorse class with methods for all the database operations you'd make from a model. The model has essentially proxy methods that call out to the model's adapter.

Under the hood something like:
```js
class Model {
  static adapter = PostgresAdapter.create(...)
  
  static all (options) {
    const results = await this.adapter.all(Model, options)
    return collection.create(results)
  }
}